src/test/java/com/iota/iri/network/TransactionRequesterTest.java

```diff
@@ -76,6 +79,48 @@ public void instance() throws Exception {
 
     }
 
+    @Test
+    public void popEldestTransactionToRequest() throws Exception {
+        TransactionRequester txReq = new TransactionRequester(tangle, snapshotProvider, mq);
+        // Add some Txs to the pool and see if the method pops the eldest one
+        Hash eldest = TransactionViewModelTest.getRandomTransactionHash();
+        txReq.requestTransaction(eldest, false);
+        txReq.requestTransaction(TransactionViewModelTest.getRandomTransactionHash(), false);
+        txReq.requestTransaction(TransactionViewModelTest.getRandomTransactionHash(), false);
+        txReq.requestTransaction(TransactionViewModelTest.getRandomTransactionHash(), false);
+
+        txReq.popEldestTransactionToRequest();
+        // Check that the transaction is there no more
+        assertEquals(false, txReq.isTransactionRequested(eldest, false));
+    }
+
+    @Test
+    public void transactionRequestedFreshness() throws Exception {
+        // Add some Txs to the pool and see if the method pops the eldest one
+        ArrayList<Hash> eldest = new ArrayList<Hash>(Arrays.asList(
+                TransactionViewModelTest.getRandomTransactionHash(),
+                TransactionViewModelTest.getRandomTransactionHash(),
+                TransactionViewModelTest.getRandomTransactionHash()
+        ));
+        TransactionRequester txReq = new TransactionRequester(tangle, snapshotProvider, mq);
+        int capacity = TransactionRequester.MAX_TX_REQ_QUEUE_SIZE;
```

Comment:
There is a thumb rule that we should declare a local variable as soon as it is being used.
Not going to insist on it now, just that you know for next time.

Change it only if you feel like it

Reply:
Since in java we have a heavily optimized compiler readability comes first
For next time :-)